lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java

```diff
@@ -427,5 +438,70 @@ public void testMultiFieldShouldReturnPerFieldBooleanQuery() throws Exception {
       analyzer.close();
     }
   }
+
+  public void testCustomFrequecy() throws IOException {
+    // define an analyzer with delimited term frequency, e.g. "foo|2 bar|3"
+    Analyzer analyzer = new Analyzer() {
+
+      @Override
+      protected TokenStreamComponents createComponents(String fieldName) {
+        MockTokenizer tokenizer = new MockTokenizer(MockTokenizer.WHITESPACE, false, 100);
+        MockTokenFilter filt = new MockTokenFilter(tokenizer, MockTokenFilter.EMPTY_STOPSET);
+        return new TokenStreamComponents(tokenizer, foo(filt));
+      }
+
+      TokenStream foo(TokenStream input) {
```

Comment:
Maybe give this method a better name?  Maybe `addCustomTokenFilter` or something?

Reply:
My bad, forgot this